### ISSUE TITLE
Old Checkpoint files no longer compatible with new firedrake

### ISSUE THREAD

**Issue by SamAhnert** [dynamicslab/hydrogym#106]
I'd like to document that in moving away from pickle (see [this commit](https://github.com/firedrakeproject/firedrake/commit/2d5aba6620638f5b5c84e7a10bc13b40c6f84f54)), the firedrake team rendered data generated and checkpointed on the old version incompatible with the new new version if I'm reading [this thread](https://github.com/firedrakeproject/firedrake/issues/2645) correctly. 

This renders our tutorial notebooks unusable with the new version of firedrake, and I think the easiest solution is to regenerate the data (especially since pickle is quite fickle with the way it ties itself to versions of packages etc.). I'n not sure how much of a priority we want this to be since it is an issue that would prevent any newcomers from being able to run our tutorials. Not necessarily a bug with the package itself, but this might deter folks interested in HydroGym unfortunately.

**Comment by ludgerpaehler** [comment]
Probably for the best if we regenerate the tutorials tbf (and expand them slightly..)

**Comment by SamAhnert** [comment]
I agree and we could include loading checkpoints for that (although maybe expanding/refactoring is a separate discussion?). I think that the checkpoints not working is an immediately pertinent issue that's worth addressing in the short-term. I can start running simulations to generate checkpoints.
